### PR TITLE
feat: npm config (.npmrc)

### PR DESCRIPTION
**Pull request by hoegertn**
Support writing a project local `.npmrc` file. E.g. this could be used to configure mirrors for all packages or for specific scopes.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by hoegertn** [comment]
I have no idea why the `new` test fails. I did not change this.

**Comment by Chriscbr** [comment]
@hoegertn Looks like a flakey test that I updated not too long ago. I'll try to submit a PR to fix it. But I think it should be safe to merge your PR. :-)

**Review by eladb — CHANGES_REQUESTED** [review]
Here's a proposed alternative API that is more inline with projen's (undocumented :-)) API standards:

```ts
export interface NpmRcOptions {
  /**
   * NPM registry to use
  readonly registry?: string;
}

export interface NpmRc extends Component {
  private readonly obj: any;

  constructor(project: NodeProject, options: NpmRcOptions = {}) {
    super(project, { obj });

    if (options.registry) {
      this.addRegistry(options.registry);
    }
  }

  public addRegistry(url: string, scope?: string) {
    const key = scope ? `${scope}:registry` : 'registry';
    this.addConfig(key, url);
  }

  public addConfig(name: string, value: string) {
    this.obj[name] = value;
  }
}
```

Also:
* Make sure to include docstrings on all public APIs
* Add unit tests

**Comment by hoegertn** [comment]
@eladb I changed the API a bit

File: src/javascript/node-config.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+   *
+   * @default - use npmjs default registry
+   */
+  registry?: string;
+}
+
+/**
+ * File representing the local NodeJS config in .npmrc
+ */
+export class NodeConfig extends Component {
```
Comment (eladb) [inline src/javascript/node-config.ts:20]: Rename to `NpmConfig` or `NpmRc`

File: src/javascript/node-config.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    }
+  }
+
+  /**
+   * configure a scoped registry
+   *
+   * @param scope the scope the registry is used for
+   * @param url the URL of the registry to use
+   */
+  public configureRegistry(scope: string, url: string) {
```
Comment (eladb) [inline src/javascript/node-config.ts:44]: I preferred the previous version so it's possible to specify registry after initialization:

```ts
addRegistry(url: string, scope?: string)
```

File: src/javascript/node-config.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import { Component } from '../component';
+import { IniFile } from '../ini';
+import { NodeProject } from '../node-project';
+
+/**
+ * Options to configure the local NodeJS config
+ */
+export interface NodeConfigOptions {
+  /**
+   * URL of the registry mirror to use
```
Comment (eladb) [inline src/javascript/node-config.ts:10]: Mention `addRegistry` here

**Review by eladb — CHANGES_REQUESTED** [review]

**Review by eladb — APPROVED** [review]
Looks good!

**Comment by hoegertn** [comment]
How do we proceed with the failing test? Is this fixed?

**Comment by eladb** [comment]
> How do we proceed with the failing test? Is this fixed?

I believe it was fixed (#821) by @Chriscbr

**Comment by eladb** [comment]
Did you merge the latest from main?

**Comment by hoegertn** [comment]
Merged it. I was confused as I expected GitHub to show an update branch button...

**Pull request merged**